Sage/carbon
# fix(link): change display to inline so that text can wrap

Link components currently use `display: inline-block` for both the anchor element itself and one of the containing components. This means that when the text of a link is long enough to wrap onto a new line, the entire link element is moved to the next line, creating unnecessary blank space. This PR fixes that by removing the `display: inline-block`, except when used in a MenuItem - that component will never be used inline and the inline-block display is needed for the background color on hover to fill the entire vertical space.

fix #5774

### Proposed behaviour

Links should allow text to wrap in the middle of the component's text.



### Current behaviour

If the full link text does not fit in the available space, the entire link is moved onto a new line.



### Checklist




#### QA


### Additional context



### Testing instructions



The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

Note - this sandbox has a slight difference from the one in the Github issue, in that an additional space has been added (rather than just a newline) before the <Link> component. Without this, the bug is still fixed but there is no space between the link text and the previous word. https://codesandbox.io/s/confirm-modal-forked-09n7p7?file=/src/index.js

Also test any relevant stories for components that use Link internally, to ensure that nothing has changed that shouldn't have. The following components are the ones that use Link:
- Heading (if `backLink` prop is used)
- Menu (for all menu items - including those that open submenus, as well as the individual items in submenus)
- Pager (the "links" for moving between pages)
